test/modern_bpf/test_suites/syscall_exit_suite/execve_x.cpp

```diff
@@ -0,0 +1,234 @@
+#include "../../event_class/event_class.h"
+#include "../../flags/flags_definitions.h"
+
+#if defined(__NR_execve) && defined(__NR_getrlimit) && defined(__NR_capget)
+
+#include <sys/resource.h> // to get rlimit_resource
+
+#define MAX_PATH 4096
+
+TEST(SyscallExit, execveX)
+{
+	auto evt_test = new event_test(__NR_execve, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	/* We extract some info from proc before calling the syscall, in this way we are sure that
+	 * the numbers that we will extract	in the BPF are surely greater or equal than the ones extracted now.
+	 */
+	pid_t process_pid = ::getpid();
+	char path_to_read[MAX_PATH];
+
+	/*
+	 * Read some info regarding the stats of the process from `/proc/[pid]/stat`
+	 */
+	snprintf(path_to_read, sizeof(path_to_read), "/proc/%d/stat", process_pid);
+	FILE* stat = fopen(path_to_read, "r");
+	if(stat == NULL)
+	{
+		FAIL() << "'fopen /proc/[pid]/stat' must not fail: (" << errno << "), " << strerror(errno) << std::endl;
+	}
+
+	int tty = 0;
+	unsigned long minflt = 0;
+	unsigned long majflt = 0;
+	pid_t ppid = 0; /* The PID of the parent of this process. */
+	pid_t pgid = 0; /* The process group ID of the process. */
+
+	/* we could get the filename of the executable (`comm`) from proc, but it is returned
+	 * in parentheses, so for example "(bpf_test)", so we prefer to use our macro `TEST_EXECUTABLE_NAME`
```

Comment:
It could be possible to use `"(%s)"` fmt to read `comm`, however, `%s` will include `)` that needs to be corrected.

Reply:
yeah, not sure what is the best way to do it :thinking:

Reply:
I also do not have a strong opinion.. so maybe, let's leave it as for now.